File: src/main/java/scrumweb/dto/fieldcontent/CheckBoxContainerContentDto.java
package scrumweb.dto.fieldcontent;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import scrumweb.dto.projectfield.CheckBoxDto;

import java.util.Set;

@NoArgsConstructor
@Getter @Setter
public class CheckBoxContainerContentDto extends FieldContentDto {

    private Set<CheckBoxDto> elements;

    public CheckBoxContainerContentDto(Long projectFieldId, String projectFieldName, Set<CheckBoxDto> elements) {
        super(projectFieldId, projectFieldName);
        this.elements = elements;
    }
}

File: src/frontend/src/app/model/fields-content/CheckBoxContainerContentDto.ts
import { CheckBoxDto } from './CheckBoxDto';
import { FieldContentDto } from './FieldContentDto';

export class CheckBoxContainerContentDto extends FieldContentDto {
    elements: Array<CheckBoxDto> = [];

    constructor(projectFieldId: number, projectFieldName: string, elements: Array<CheckBoxDto>) {
        super(projectFieldId, projectFieldName);
        this.elements = elements;
    }
}
